### ISSUE TITLE
Malfunction of SPARQL query containing non-ASCII characters

### ISSUE THREAD

**Issue by mchlrch** [zazuko/trifid#38]
I have a SPARQL query with german Umlaut characters (äöü) in string literals. Running the query directly against a local triplestore works as expected. The encoding of those non-ASCII characters gets broken if Trifid is proxying the query.

Notice that non-ASCII characters from the data in the triplestore are returned just fine in the response. The encoding seems to break for the request (on the way in), not for the response.

See the last line in the snippet below: `bÃ¤r,Löwenbräu` should be `bär,Löwenbräu`

I tested this with Stardog as triplestore on Ubuntu.

```
$ cat data.ttl 
BASE <http://example.org/>
PREFIX schema: <http://schema.org/>

<p1>
    a schema:Person;
    schema:familyName "Löwenbräu".

$ cat query5.rq 
PREFIX schema: <http://schema.org/>

SELECT ?foo ?name
WHERE {
  ?p a schema:Person;
    schema:familyName ?name.
  BIND("bär" AS ?foo )
}
LIMIT 10

$ stardog-admin db create -n umlaut_test1 data.ttl
Bulk loading data to new database umlaut_test1.
Loaded 2 triples to umlaut_test1 from 1 file(s) in 00:00:00.098 @ 0.0K triples/sec.
Successfully created database 'umlaut_test1'.

$ curl -i -H "Accept: text/csv"  --data-urlencode query@query5.rq http://admin:admin@localhost:5820/umlaut_test1/query
HTTP/1.1 200 OK
Connection: keep-alive
SD-Query-Id: 10
Transfer-Encoding: chunked
Content-Type: text/csv
Date: Fri, 04 May 2018 11:37:21 GMT

foo,name
bär,Löwenbräu

$ curl -i -H "Accept: text/csv"  --data-urlencode query@query5.rq http://localhost:8080/query
HTTP/1.1 200 OK
connection: close
sd-query-id: 11
transfer-encoding: chunked
content-type: text/csv
date: Fri, 04 May 2018 11:37:30 GMT
access-control-allow-origin: *
cache-control: public, max-age=120
Vary: Accept, Accept-Encoding

foo,name
bÃ¤r,Löwenbräu
```

**Comment by bergos** [comment]
It was already working with `queryOperation` set to `postQueryUrlencoded`. Non-ASCII chars were encoded with the standard URL encoding. But with `postQueryDirect` no charset was defined, which get's interpreted differently by different endpoint implementations. Fuseki interpreted the content already as UTF-8. With [this commit to sparql-http-client](https://github.com/zazukoians/sparql-http-client/commit/8fc7eee187d855be38bd160d01fabcca5ee0ee57) the charset is included in the `Content-Type` header, which should fix the problem for all other endpoint implementations.

**Comment by mchlrch** [comment]
This got fixed with version 1.0.3 of `sparql-http-client`.
